src/Model/Pingback.php

```diff
@@ -20,59 +22,103 @@ public function __construct(
 
     public function pingback($getData)
     {
-        $this->_helper->getInitConfig();
+        if (empty($getData['goodsid'])) {
+            return "Order invalid !";
+        }
+        $orderIncrementId = $getData['goodsid'];
+        $orderModel = $this->_objectManager->get('Magento\Sales\Model\Order');
+        $orderModel->loadByIncrementId($orderIncrementId);
+        $method = $orderModel->getPayment()->getMethodInstance()->getCode();
+
+        if ($method == self::PWLOCAL_METHOD) {
+            $this->_helper->getInitConfig();
+        } else {
+            $this->_helper->getInitBrickConfig(true);
+        }
+
         $pingback = new \Paymentwall_Pingback($getData, $_SERVER['REMOTE_ADDR']);
```

Comment:
Please create a new function to get real client IP.
It will support the merchants who use proxy services like CloudFlare, Nginx, Varnish...